### PR TITLE
feat: Add `alwaysRender` prop to force offscreen rendering

### PR DESCRIPTION
**Pull request by jwb**
Add a new option that suppresses the optimization of skipping rendering when the component has no space to display.

### CODE REVIEW

File: index.jsx [hunk 1]
```diff
@@ -169,14 +172,15 @@ export default function Dimensions ({
 
       render () {
         const {containerWidth, containerHeight} = this.state
-        if (this._parent && !containerWidth && !containerHeight) {
-          // only trigger a warning about the wrapper div if we already have a reference to it
+        const renderComponent = containerWidth || containerHeight || alwaysRender
```
Comment (evocateur) [inline index.jsx:175]: Don't we still need to care about `this._parent`?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "react-dimensions",
-  "version": "2.0.0-alpha1",
+  "name": "@zillow/react-dimensions",
```
Comment (evocateur) [inline package.json:2]: I would rather these changes were in a separate PR (or just commit them directly to master, I assume we're not going to push anything upstream anymore?)

Comment (evocateur) [reply package.json:2]: (the name and version of a fork don't _need_ to be versioned, of course)

**Review by evocateur — COMMENTED** [review]
Looks okay, a bit weird that the upstream commits are identified in this PR, yet not present in the diff...

File: package.json (hunk 2 above)
Comment (stonebk) [reply package.json:2]: master has been abandoned -- perhaps we can take ownership of it at some point

File: index.jsx (hunk 1 above)
Comment (stonebk) [reply index.jsx:175]: Would prefer `shouldRender` naming

Comment (jwb) [reply index.jsx:175]: I dunno, if you don't like `alwaysRender`, how about `forceRender`? To me, if `shouldRender` is false, it means to suppress rendering, not to allow optimization. Another name could be `suppressOptimization`.

Comment (stonebk) [reply index.jsx:175]: I'm OK with `alwaysRender`, but `renderComponent` -> `shouldRender`

**Comment by jwb** [comment]
Oh gee. I started my branch from the wrong point. Thanks for catching that @evocateur! I'll try again.

**Comment by jwb** [comment]
As pointed out, this branch includes commits whose changes have been removed. Closing this PR and starting again in PR #3.

**Comment by evocateur** [comment]
Generally speaking, you don't have to close a PR just to do a force push. Now we have no conversation history in the new PR... /shrug

**Comment by jwb** [comment]
Sorry. I'm guess I'm too leery of force-push. I'll try to be more brave in the future.

**Pull request closed without merging**